### PR TITLE
Hyperparameter defaults + master update

### LINKED ISSUE #193: Validate default hyperparameter grids for all methods
## Tasks

- [x] validate that all default hyperparameter grids are sane and decent
- [x] In `advanced.ensemble_classification()` set `self.logistic_regression(randomized_search=True)` rather than `False`

### PR DESCRIPTION
**Pull request by Aylr**
Sorry, I made the mistake of updating this branch with master, so this PR closes #193 and #214 

- [ ] look at the hyperparameters
- [ ] the master update is ugly, but it is mostly documentation updates

### CODE REVIEW

File: healthcareai/develop_supervised_model.py [hunk 1]
```diff
@@ -268,7 +265,6 @@ def random_forest_classifier(self, trees, scoring_metric='roc_auc', hyperparamet
         default hyperparameter grid.
         """
         if hyperparameter_grid is None:
-            # TODO add sensible hyperparameter grid
             max_features = helpers.calculate_random_forest_mtry_hyperparameter(len(self.X_test.columns),
                                                                                self.model_type)
             hyperparameter_grid = {'n_estimators': [10, 50, 200], 'max_features': max_features}
```
Comment (mmastand) [inline healthcareai/develop_supervised_model.py]: Let's do 50, 100, 250

File: healthcareai/develop_supervised_model.py [hunk 2]
```diff
@@ -228,9 +227,7 @@ def knn(self, scoring_metric='roc_auc', hyperparameter_grid=None, randomized_sea
         hyperparameter grid.
         """
         if hyperparameter_grid is None:
-            # TODO add sensible KNN hyperparameter grid
-            neighbor_list = list(range(10, 26))
-            hyperparameter_grid = {'n_neighbors': neighbor_list, 'weights': ['uniform', 'distance']}
+            hyperparameter_grid = {'n_neighbors': list(range(5, 26)), 'weights': ['uniform', 'distance']}
```
Comment (mmastand) [inline healthcareai/develop_supervised_model.py]: I don't know knn at all, unfortunately.

File: healthcareai/develop_supervised_model.py [hunk 3]
```diff
@@ -209,7 +208,7 @@ def linear_regression(self, scoring_metric='neg_mean_squared_error', hyperparame
         hyperparameter grid.
         """
         if hyperparameter_grid is None:
-            # TODO sensible default hyperparameter grid
+            hyperparameter_grid = {"fit_intercept": [True, False]}
             pass
```
Comment (mmastand) [inline healthcareai/develop_supervised_model.py]: L1 vs. L2 regularization?

File: healthcareai/develop_supervised_model.py (hunk 2 above)
Comment (mmastand) [inline healthcareai/develop_supervised_model.py]: Ok, I read up and these look ok. Let's keep an eye on them though.

File: healthcareai/develop_supervised_model.py (hunk 3 above)
Comment (Aylr) [inline healthcareai/develop_supervised_model.py]: Noted in #267

File: healthcareai/develop_supervised_model.py (hunk 1 above)
Comment (Aylr) [inline healthcareai/develop_supervised_model.py]: Noted in #268

**Pull request merged**